### PR TITLE
DrivAerML: T_max=40 + lr=4.8e-4 MSE-only (schedule+LR compound)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Your T_max=40 + metric-aware w=0.05 at lr=5e-4 gave val=3.651% — only 0.029pp short of the 3.622% baseline. lr=4.8e-4 gave the best full-eval TEST in the programme (4.117%). Combining T_max=40 + lr=4.8e-4 MSE-only could tip the balance — the slightly longer cycle + lower LR may find a deeper basin.

## Command
```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py --dataset drivaerml --optimizer adamw --lr 4.8e-4 --cosine-t-max 40 --grad-clip 0.5 --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint --wandb_group dm-tmax40-lr48e4
```
Report: best val/rel_l2_mean, epoch, W&B run ID, stability. Target: val < 3.622%

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT alphonse:

## Results

### DrivAerML: T_max=40 + lr=4.8e-4 MSE-only

| Metric | Baseline | This Run | Delta |
|--------|----------|----------|-------|
| val_primary/surface_rel_l2_pct | 3.622% (ep 874) | **3.898%** (ep 733) | +0.276pp ❌ |
| test_primary/surface_rel_l2_pct (batch-limited) | 4.631% | **4.670%** | +0.039pp ❌ |
| SOTA target | <3.71% | 3.898% | +0.188pp above SOTA ❌ |

- **Best val:** 3.898% at epoch 733 (out of 890 epochs total, 600-min timeout)
- **Best test (batch-limited):** 4.670%
- **W&B run:** `uwyig0cw` (alphonse/dm-tmax40-lr48e4)
- **Checkpoint:** `outputs/icml2026/drivaerml_senpai_transolver_best.pt` (epoch 733)

**Command used:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_TIMEOUT_MINUTES=600 python train.py --dataset drivaerml --optimizer adamw --lr 4.8e-4 --cosine-t-max 40 --grad-clip 0.5 --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint --wandb-group dm-tmax40-lr48e4 --agent alphonse --wandb-name "alphonse/dm-tmax40-lr48e4"
```

### What happened

**The hypothesis did not work.** T_max=40 + lr=4.8e-4 MSE-only reached 3.898% val, which is 0.276pp worse than the 3.622% baseline (T_max=36 + lr=5e-4). The run also failed to beat the SOTA target of 3.71%.

Key observations:
- **Convergence pattern:** The model showed steady improvement through cosine cycles with diminishing returns. Major improvements came in the first ~400 epochs, then gains became marginal (~0.02-0.04pp/cycle).
- **Late breakthrough:** After a long plateau at 3.999% (epochs 580-680), the model found a deeper basin around epoch 732, dropping to 3.898%. However, it then plateaued completely for the remaining ~160 epochs.
- **T_max=40 vs T_max=36:** The longer cosine cycle did not help — T_max=36 at lr=5e-4 reached 3.622% at the same epoch count. The gentler schedule may have been too conservative for the lower learning rate.
- **lr=4.8e-4 vs lr=5e-4:** The slightly lower LR appears to have limited the model's ability to escape local minima during warm-up phases, resulting in longer and more frequent plateaus.
- **Total training:** 890 epochs in 600 minutes.

### Suggested follow-ups

1. **T_max=36 + lr=4.8e-4:** Test whether lr=4.8e-4 works better with the proven T_max=36 schedule — isolate LR vs schedule effects.
2. **T_max=40 + lr=5e-4:** Test whether T_max=40 works better with the proven lr=5e-4 — isolate schedule vs LR effects.
3. The baseline T_max=36 + lr=5e-4 combination appears to be near-optimal for this architecture.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3388 because ADVISOR: Closing — T_max=40 + lr=4.8e-4 did not beat baseline. val=3.898% (+0.276pp worse than 3.622% baseline). The longer cosine cycle plateaued at ep732 with no further improvement through ep890. T_max=36 remains the local optimum for this LR range. Thank you alphonse..
